packages/api/endpoints/collections.js

```diff
@@ -43,14 +44,22 @@ const log = new Logger({ sender: '@cumulus/api/collections' });
  * @returns {Promise<Object>} the promise of express response object
  */
 async function list(req, res) {
+  log.trace(`list query ${JSON.stringify(req.query)}`);
   const { getMMT, includeStats, ...queryStringParameters } = req.query;
-  const collection = new Collection(
-    { queryStringParameters },
-    undefined,
-    process.env.ES_INDEX,
-    includeStats === 'true'
-  );
-  let result = await collection.query();
+  let dbSearch;
+  if (includeStats === 'true') {
```

Comment:
Not to belabor the work, but we should enable @ts-check and get the typings right for this method, especially if adding if/else logic, per our ADRs.

Reply:
I mean, we can toss out the if/else portion of that and the method should still be annotated/typed with @ts-check.

Reply:
So you're saying we should add a `// @ts-check` somewhere near the top of this file?

Reply:
@charleshuang80  yes, exactly, and then resolve the issues related to the *changing* code.   Fixing all of it is a thing we are doing as we're updating.